src/lib/Libutil/daemon_protect.c

```diff
@@ -62,15 +63,15 @@ daemon_protect(pid_t pid, enum PBS_Daemon_Protect action)
 	 */
 	snprintf(fname, MAXPATHLEN, oom_protect_new.oom_path, pid);
 	if ((fd = open(fname, O_WRONLY)) != -1) {
-		write(fd, oom_protect_new.oom_value[(int)action], sizeof(oom_protect_new.oom_value[(int)action])-1);
+		write(fd, oom_protect_new.oom_value[(int)action], strlen(oom_protect_new.oom_value[(int)action]));
```

Comment:
In addition, I think we want to open the file as O_WRONLY|O_TRUNC
Same below as well.